connstr/README.md

```diff
@@ -0,0 +1,118 @@
+
+## Connection Strings
+
+The `connectString` parameter can be one of:
+
+- An Easy Connect String
+
+   [Easy Connect strings](https://www.oracle.com/pls/topic/lookup?ctx=dblatest&id=GUID-B0437826-43C1-49EC-A94D-B650B6A4A6EE)
+   like the one shown above are most convenient to use.   The syntax allows a
+   number of options to be set without requiring an external `tnsnames.ora`
+   configuration file.   A more complex example usable with Oracle Client
+   libraries 19c or later is:
+
+   ```
+   connectString="tcps://salesserver1:1521/sales.us.example.com?ssl_server_cert_dn=\"cn=sales,cn=Oracle Context Server,dc=us,dc=example,dc=com\"&sdu=8128&connect_timeout=60"
+   ```
+
+   Common options are `connect_timeout` to return an error if connection takes
+   too long, and `expire_time` to make sure idle connections are not closed by
+   firewalls.
+
+   The technical article [Oracle Database 19c Easy Connect Plus Configurable
+   Database Connection Syntax](https://download.oracle.com/ocomdocs/global/Oracle-Net-19c-Easy-Connect-Plus.pdf)
+   contains more information.
+
+- An Oracle Net Connect Descriptor String
+
+   Full descriptors can be used, such as
+   `connectString="(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=dbhost.example.com)(PORT=1521))(CONNECT_DATA=(SERVICE_NAME=orclpdb1)))"`
+
+- An Oracle Net Service Name
+
+   Connect Descriptor Strings are commonly stored in a `tnsnames.ora` file and
+   associated with a Net Service Name.   This name can be used directly for
+   `connectString`.   For example, given a `tnsnames.ora` file with the following
+   contents:
+
+         ORCLPDB1 =
+            (DESCRIPTION =
+               (ADDRESS = (PROTOCOL = TCP)(HOST = dbhost.example.com)(PORT = 1521))
+               (CONNECT_DATA =
+                  (SERVER = DEDICATED)
+                  (SERVICE_NAME = orclpdb1)
+               )
+            )
+
+   then you could connect using `connectString=ORCLPDB1`
+
+   See "Optional Oracle Net Configuration Files" below.
+
+## Optional Oracle Net Configuration Files
+
+Optional Oracle Net configuration files are used by the Oracle Client libraries
+during the first call to `sql.Open`.   The directory containing the files can be
+specified in the `sql.Open()` data source name with the `configDir` option.
+
+The common files are:
+
+* `tnsnames.ora`: A configuration file that defines databases aliases for
+   establishing connections.
+
+* `sqlnet.ora`: A profile configuration file that may contain information on
+   features such as connection failover, network encryption, logging, and
+   tracing.   See [Oracle Net Services
+   Reference](https://www.oracle.com/pls/topic/lookup?ctx=dblatest&id=GUID-19423B71-3F6C-430F-84CC-18145CC2A818)
+   for more information.
+
+The files should be in a directory accessible to godror, not on the database
+server host.
+
+If the `configDir` connection option is not used, then a search heuristic is
+used to find the files.   The search includes:
```

Comment:
I would now tweak this to read "used to find the directory containing the files."